src/plot_api/manage_arrays.js

```diff
@@ -101,7 +101,7 @@ exports.applyContainerArrayChanges = function applyContainerArrayChanges(gd, np,
         return true;
     }
 
-    var componentNums = Object.keys(edits).map(Number).sort(),
+    var componentNums = Object.keys(edits).map(Number).sort(function(a, b) { return a - b; }),
```

Comment:
You might want to use [`Lib.sorterAsc`](https://github.com/plotly/plotly.js/blob/ce970f112a8fff0d09909529be8a713f5d4f36cc/src/lib/search.js#L59) to DRY it up :palm_tree: